apps/dropbox/src/inngest/functions/data-protection/create-sync-folders-and-files.ts

```diff
@@ -0,0 +1,91 @@
+import { inngest } from '@/inngest/client';
+import { getOrganisationAccessDetails } from '../common/data';
+import { DBXUsers, getElba } from '@/connectors';
+import { InputArgWithTrigger } from '@/inngest/types';
+import { decrypt } from '@/common/crypto';
+import { NonRetriableError } from 'inngest';
+
+const handler: Parameters<typeof inngest.createFunction>[2] = async ({
+  event,
+  step,
+}: InputArgWithTrigger<'dropbox/data_protection.folder_and_files.create.sync_page.requested'>) => {
+  const { organisationId, cursor, syncStartedAt } = event.data;
+
+  const [organisation] = await getOrganisationAccessDetails(organisationId);
+
+  if (!organisation) {
+    throw new NonRetriableError(
+      `Access token not found for organisation with ID: ${organisationId}`
+    );
+  }
+
+  const { accessToken, region } = organisation;
+
+  const token = await decrypt(accessToken);
+
+  const dbxUsers = new DBXUsers({
+    accessToken: token,
+  });
+
+  const elba = getElba({
+    organisationId,
+    region,
+  });
+
+  const team = await step.run('fetch-users', async () => {
+    return dbxUsers.fetchUsers(cursor);
+  });
+
+  if (!team) {
+    throw new Error(`Users not found for the organisation  with ID: ${organisationId}`);
+  }
+
+  const fileSyncJobs = team.members.map(({ id: teamMemberId }) => {
+    return {
+      ...event.data,
+      teamMemberId,
+    };
+  });
+
+  if (team.members.length > 0) {
+    await step.sendEvent(
+      'dropbox-sync-folder-and-files-page',
+      fileSyncJobs.map((fileSyncJob) => ({
+        name: 'dropbox/data_protection.folder_and_files.sync_page.requested',
+        data: fileSyncJob,
+      }))
+    );
+  }
+
+  if (team.hasMore) {
+    return await step.sendEvent('dropbox-create-sync-folder-and-files-page', {
+      name: 'dropbox/data_protection.folder_and_files.create.sync_page.requested',
+      data: {
+        ...event.data,
+        cursor: team.nextCursor,
+      },
+    });
+  }
+
+  await step.run('delete-objects', async () => {
+    await elba.dataProtection.deleteObjects({
+      syncedBefore: new Date(syncStartedAt).toISOString(),
+    });
+  });
+};
+
+export const createPathSyncJobs = inngest.createFunction(
```

Comment:
```suggestion
export const createPathSyncJobs = inngest.createFunction(
```
The syncjobs part could be confusing and do not really match the event triggering the function. Do you have any other name in your naming hat ?

Reply:
This naming is still an issue. What does the function is unclear ? After looking at the code it's seems to start the folders & files sync, so why not using `start-folders-and-files-sync` ?